dvc/proc/manager.py

```diff
@@ -63,17 +65,29 @@ def spawn(self, args: Union[str, List[str]], name: Optional[str] = None):
             pid,
         )
 
-    def send_signal(self, name: str, signal: int):
+    def send_signal(self, name: str, sig: int):
         """Send `signal` to the specified named process."""
-        raise NotImplementedError
-
-    def kill(self, name: str):
-        """Kill the specified named process."""
-        raise NotImplementedError
+        process_info = self[name]
+        if sys.platform == "win32":
+            if sig not in (
+                signal.SIGTERM,
+                signal.CTRL_C_EVENT,
+                signal.CTRL_BREAK_EVENT,
+            ):
+                raise ValueError(f"Unsupported signal: {sig}")
+        if process_info.returncode is None:
+            os.kill(process_info.pid, sig)
```

Comment:
We need to account for potential `OSError`s raised on `os.kill()`. I think in general we do want to re-raise any exceptions, but there is at least one special case we should account for:

If the spawned daemon process crashes (or is explicitly killed by some other event, like the user running `kill -9` or a system crash/shutdown), this PID will no longer exist. For this error case we should catch `ProcessLookupError`, and then explicitly set `process_info.returncode` to `-1` and update the info file (to indicate that the process was closed due to some unknown error), so that we don't try to treat it as a running process again in subsequent calls. After we've done that, we can then re-raise the original `ProcessLookupError` exception up the call stack.